### PR TITLE
MBS-9905: Tipeee Broken URL validation

### PR DESCRIPTION
**Pull request by Ge0rg3**
## [MBS-9905](https://tickets.metabrainz.org/browse/MBS-9905)
When inputting a Tipeee URL that begins with a language-related subdomain, i.e...

https://en.tipeee.com/borisdalstein

...the URL is not recognised as tipeee. This is the case with all 5 languages that the Tipeee website supports: **en**, **it**, **fr**, **de** & **es**.

An example of a _working_ URL would be:
https://www.tipeee.com/borisdalstein

----

image

### CODE REVIEW

File: root/static/scripts/edit/URLCleanup.js [hunk 1]
```diff
@@ -2025,7 +2025,7 @@ const CLEANUPS = {
       new RegExp("^(https?://)?(www\\.)?flattr\\.com/profile/[^/?#]", "i"),
       new RegExp("^(https?://)?(www\\.)?patreon\\.com/[^/?#]", "i"),
       new RegExp("^(https?://)?(www\\.)?paypal\\.me/[^/?#]", "i"),
-      new RegExp("^(https?://)?(www\\.)?tipeee\\.com/[^/?#]", "i"),
+      new RegExp("^(https?://)?(www\.)?(?:(?:fr|en|de|es|it)\.)?tipeee\.com/[^/?#]", "i"),
```
Comment (yvanzo) [inline root/static/scripts/edit/URLCleanup.js]: It is probably not needed to match `www.en.tipeee` but it's still better than the previous version of this RegExp.

Comment (Ge0rg3) [reply root/static/scripts/edit/URLCleanup.js]: Okay, would it be better as:
[...] https?://?`((?:www|fr|en|de|es|it)\.)`?tipeee\.com [...]

Comment (yvanzo) [reply root/static/scripts/edit/URLCleanup.js]: Yes, but if you want it to keep working for any additional language in the future, just use `(?:[^/]+\\.)?tipeee\\.com` instead, that will match any subdomain.

Note that dot should be escaped twice `\\.` here.

Comment (Ge0rg3) [reply root/static/scripts/edit/URLCleanup.js]: That sounds best. I'll commit that this evening :)

**Comment by yvanzo** [comment]
I altered https://github.com/metabrainz/musicbrainz-server/commit/1c3e68855b0c4bd1b5c0662eafac862c3b722f9a commit message so as to:
 * reliably refer to the ticket key rather than to refer to the obsolete commit id;
 * follow formatting standards (first line by 50 characters, other lines wrapped to 68 characters).

**Comment by yvanzo** [comment]
Thank you for this patch!

**Pull request merged**